src/MPAGSCipher/PlayfairCipher.cpp

```diff
@@ -0,0 +1,210 @@
+            //and then use something like the insert iterator to stick it in
+            //unsure how to proceed with iterators so for now going use standard for loops etc
+            //also time constraints
+            interstring += outputText.at(0);
+            for (std::string::size_type i = 1; i < outputText.length(); i++) {
+                if (outputText.at(i - 1) == outputText.at(i)) {
+                    if (outputText.at(i) == 'X') {
+                        interstring += "Q";
+                    } else {
+                        interstring += "X";
+                    }
+                }
+                interstring += outputText.at(i);
+            }            
+            outputText = interstring;    //can replace this somehow so we dont need to deal with interstring
+
+            if (outputText.length() % 2 != 0) {
+                outputText += "Z";
+            }
+
+            break;
+
+        case CipherMode::Decrypt:
+            break;
+    }
+
+
+    // use text to find coordinate, apply rules, and return the letter
+    auto charToCoordsIter0 = letcoordmap_.find('A');
+    auto charToCoordsIter1 = letcoordmap_.find('B');
+    auto coordsToCharIter0 = coordletmap_.find(std::make_pair(0, 0));
+    auto coordsToCharIter1 = coordletmap_.find(std::make_pair(0, 0));
+
+    int x0{0};
+    int y0{0};
+    int x1{0};
+    int y1{0};
+    int x0f{0};
+    int y0f{0};
+    int x1f{0};
+    int y1f{0};
+
+    //loop over inputs in digraphs
+    interstring = "";
+
+    for (std::string::size_type i = 0; i < outputText.length(); i += 2) {
+        charToCoordsIter0 = letcoordmap_.find(outputText.at(i));
+        charToCoordsIter1 = letcoordmap_.find(outputText.at(i + 1));
+
+        //find the coordinates in grid of each diagram
+        x0 = (*charToCoordsIter0).second.first;
+        y0 = (*charToCoordsIter0).second.second;
+        x1 = (*charToCoordsIter1).second.first;
+        y1 = (*charToCoordsIter1).second.second;
+
+        x0f = x0;
+        x1f = x1;
+        y0f = y0;
+        y1f = y1;
+
+        //apply rules to get new coordinates
+        switch (cipherMode) {
+            case CipherMode::Encrypt:
+                if (y0 == y1) {    //rule1
+                    x0f = (x0 + 1) % grid_size_;
+                    x1f = (x1 + 1) % grid_size_;
+                } else if (x0 == x1) {    //rule 2
+                    y0f = (y0 + 1) % grid_size_;
+                    y1f = (y1 + 1) % grid_size_;
+                } else {    //rule 3
+                    x0f = x1;
+                    x1f = x0;
+                }
+                break;
+            case CipherMode::Decrypt:
+                if (y0 == y1) {
+                    //(b + (a%b)) % b to handle negative wrap arounds
+                    x0f = (grid_size_ + (x0 - 1) % grid_size_) % grid_size_;
+                    x1f = (grid_size_ + (x1 - 1) % grid_size_) % grid_size_;
```

Comment:
You don't want the modulus of the (x - 1) term, that could cause problems:
```suggestion
                    x0f = (grid_size_ - 1 + x0) % grid_size_;
                    x1f = (grid_size_ - 1 + x1) % grid_size_;
```